Bug Fix: Teleporting & Ability use while mounted

kirmmin opened the pull request:
Fixes #117 
Fixes #118

ahoiahoi commented on Source/NexusForever.WorldServer/Game/Entity/UnitEntity.cs:
```diff
@@ -73,6 +73,17 @@ public void CastSpell(SpellParameters parameters)
             if (parameters == null)
                 throw new ArgumentNullException();
 
+            if (GetType() == typeof(Player))
```
might want to use "if (this is Player player)" instead

Fabi commented on Source/NexusForever.WorldServer/Game/Entity/Player.cs:
```diff
@@ -536,6 +538,42 @@ public void TeleportTo(WorldEntry entry, Vector3 vector, uint instanceId = 0u, u
+        {
+            if (VehicleGuid != 0u)
+            {
+                Vehicle vehicle = GetVisible<Vehicle>(VehicleGuid);
```
Possible null reference?

kirmmin commented:
Rebased today to the latest `develop` branch.

Went through the packets today and discovered that there was an inconsistency with retail sniffs when dismounting, specifically around the fact that the EntityCommand packet update was not sent on Retail.
Removing that prevented the client "responding" to the EntityCommand instruction with it's own request.

This is the line I removed: https://github.com/NexusForever/NexusForever/pull/158/files#diff-c73a1edb9532580b34666eac934fb1daL234

I tested Dismounting, Dismount from Teleport, Logging out, and ALT+F4-ing client. None of these events triggered the exception we were seeing before (or other exceptions for that matter). I tested this with 2 clients open, as well, and was able to see the appropriate updates to both clients, so I assume the Broadcast was just an oversight. This may well be finally resolved :)

Removing WIP
